src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -599,6 +599,15 @@ public String getStatus() {
         return statusMessage == null ? "" : statusMessage.getText();
     }
 
+    public String[] getStatuses() {
```

Comment:
- Use lists.
- Rename `getStatuses()` to `getTextsForAllStatusMessagesToUser()`. If you want to, you can make this more concise, maybe something like renaming `statusMessagesToUser` to `userStatusMessages`.
- Remove `getStatus()`, what is that method supposed to do?

Reply:
>Remove `getStatus()`, what is that method supposed to do?

The reason I haven't removed `getStatus()` yet, is because `verifyStatus(final String expectedStatus)` still uses verify status, and as `getStatuses()` returns list instead of a String, I had to change the implementation of the `verifyStatus` method. So, I wanted to get a review first before modifying existing methods.

Reply:
> and only keep the methods verifyStatus(final String... expectedStatus) and getStatuses()
That was my point, sorry I didn't make it clear.

Reply:
Okay. understood. Will make the changes.

Reply:
@LiHaoTan 
>Remove `getStatus()`, what is that method supposed to do?

Removed `verifyStatus(final String expectedStatus)`, but decided against removing `getStatus()` as there are 13 more instances where it is called directly and removing it and only keeping `getStatuses()` will mean modifying every call to `getStatus()`. (And as one returns a list and the other a string, this will mean a lot of refactoring which I don't know will be worth it?). What do you think?

Reply:
Yes, you need to remove `getStatus` do the necessary changes. `getStatus` doesn't mean anything appropriate.

Note that if you use `userStatusMessages` in place of `statusMessagesToUser` you need to replace every inconsistency out there, including `statusToAdmin` and that is not very trivial.

Here's what happened:
- When multiple status messages were supported, the corresponding tests were not modified.
- Nobody since has ever tested multiple status messages, so the broken tests are finally discovered

Reply:
@LiHaoTan 
>Yes, you need to remove getStatus and make the necessary changes. getStatus doesn't mean anything appropriate.<bR>
Note that if you use userStatusMessages in place of statusMessagesToUser you need to replace every inconsistency out there, including statusToAdmin and that is not very trivial.

If I may suggest something. In my opinion I think this will need a lot of refactoring. So why don't I create a new issue for all the changes you suggested and push the changes to that issue After this gets merged to keep the scope of the PR within just the new status message? Will that be a good solution?

Reply:
I think you got the order wrong. This issue depends on the testability of status messages.

If you were to fix the testability of status messages first, that other issue needs to be merged before we can merge this PR.

So if you want to create a new issue, feel free to but you are going to have to fix the testability first before we can get to what we want to do in this issue.

Reply:
In that case, it will be easier for me to make the changes in this PR only, so that I can test the changes I will be making.